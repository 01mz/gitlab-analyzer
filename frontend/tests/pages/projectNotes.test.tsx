import React from 'react';
import {render} from '@testing-library/react';
import Index from '../../pages/project/[projectId]/notes';

describe("Project Notes", () =>{
    const useRouter = jest.spyOn(require('next/router'), 'useRouter');
    const mockUseEffect = jest.spyOn(React, 'useEffect')
    const mockAxios = jest.spyOn(require('axios'), 'get');

    beforeEach(()=>{
        useRouter.mockClear();
        mockUseEffect.mockClear();
        mockAxios.mockClear();
    })

   it("Snapshot projectId", () => {
        useRouter.mockImplementationOnce(() => ({
            query: { projectId: 'TestId', startDateTime: '2020-08-22T15:40-05:00', endDateTime:'2021-08-22T15:40-05:00' },
        }));
        const { container } = render(
            <Index />
        )
        expect(container).toMatchSnapshot();

    })
    it("Test useEffect", ()=>{
        useRouter.mockImplementationOnce(() => ({
            query: { projectId: 'TestId', startDateTime: '2020-08-22T15:40-05:00', endDateTime:'2021-08-22T15:40-05:00' },
        }));
        render(<Index />);
        expect(mockUseEffect).toBeCalled();

    })

    it('Test axios',()=>{
        useRouter.mockImplementationOnce(() => ({
            query: { projectId: 'TestId', startDateTime: '2020-08-22T15:40-05:00', endDateTime:'2021-08-22T15:40-05:00' },
        }));
        render(<Index />);
        expect(mockAxios).toHaveBeenCalledTimes(2);
    })


})