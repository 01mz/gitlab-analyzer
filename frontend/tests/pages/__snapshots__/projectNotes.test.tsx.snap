// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project Notes Snapshot projectId 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-4"
  >
    <div
      class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style="width: 7em; height: 7em;"
    >
      <svg
        class="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </div>
  </div>
</div>
`;
